src/main/kotlin/ui/settings/SettingsPresenter.kt

```diff
@@ -33,94 +46,163 @@ class SettingsPresenter(private val view: SettingsView,
     fun onViewCreated() = view.setScreenElementDetailsEnabled(false)
 
     private fun resetToInitialSettings() {
-        screenElements.clear()
-        initialSettings.screenElements.mapTo(screenElements) { it.copy() }
+        categories.clear()
+        initialSettings.categories.mapTo(categories) { it.copy() }
+        currentSelectedCategory = initialSettings.currentlySelectedCategory
```

Comment:
This other issue with persisting state even when cancel is clicked originates in here. The problem is that you are not creating deep copies of the objects from `initialSettings` and you are modifying them later, which triggers persisting mechanism.